core/templates/services/html-length.service.ts

```
@@ -50,51 +65,129 @@ export class HtmlLengthService {
     }
     const sanitizedHtml = this.sanitizer.sanitize(
       SecurityContext.HTML, htmlString) as string;
-    let totalWeight = this.calculateBaselineLength(sanitizedHtml);
+    // Identify custom tags using regex on the original HTML string.
+    const customTags = htmlString.match(CUSTOM_TAG_REGEX);
+    let customTagsLength = (customTags) ? customTags.length : 0;
+
+    /* CustomTagsLength is being passed to calculateBaselineLength()
```

I don't understand this comment. What do you mean by "if we use non-interactive RTE" and why would the block raise an error? A comment on the "outer scope" of the code should not make references to implementation details of the inner scope.

Perhaps you could just replace each custom tag in the string with `''` before passing it into calculateBaselineLength()? Then you don't need the extra argument.

---

@seanlip Comment has been Removed and below is a detailed explanation,

- Try-catch block in ``calculateBaselineLength()`` will raise an error, if empty string ( ``' '`` ) is passed to that.
- Cases when can a  ``' '`` be passed to ``calculateBaselineLength()`` :
   - Normal string is passed in ``computeHtmlLength()``, it will sanitize that and convert that into ``' '``.
   - Custom tags are passed in ``computeHtmlLength()`` except link tag and concept card tag,  it will sanitize that and convert that into ``' '``.

Initially, when I wrote this try-catch block it was to raise an error for a normal string, therefore overlooked the custom tag condition which I caught in manual testing.

Now we have to raise an error if the normal string is passed and also prevent raising an error when custom tags are used, so therefore logic applied is:
- If we use custom tags, then the ``customTags`` array will have a certain length which will work as a check for an empty string that is passed because of custom tags or not! Thanks

---

Thanks -- see my comment below though about why we even need to throw an error in the first place. If we don't need that requirement then perhaps you can simplify this as discussed.